src/Compilers/CSharp/Portable/Symbols/PublicModel/NamespaceSymbol.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel
+{
+    internal sealed class NamespaceSymbol : NamespaceOrTypeSymbol, INamespaceSymbol
+    {
+        private readonly Symbols.NamespaceSymbol _underlying;
+
+        public NamespaceSymbol(Symbols.NamespaceSymbol underlying)
+        {
+            Debug.Assert(underlying is object);
+            _underlying = underlying;
+        }
+
+        internal override CSharp.Symbol UnderlyingSymbol => _underlying;
+        internal override Symbols.NamespaceOrTypeSymbol UnderlyingNamespaceOrTypeSymbol => _underlying;
+        internal Symbols.NamespaceSymbol UnderlyingNamespaceSymbol => _underlying;
+
+        bool INamespaceSymbol.IsGlobalNamespace => _underlying.IsGlobalNamespace;
+
+        NamespaceKind INamespaceSymbol.NamespaceKind => _underlying.NamespaceKind;
+
+        Compilation INamespaceSymbol.ContainingCompilation => _underlying.ContainingCompilation;
+
+        ImmutableArray<INamespaceSymbol> INamespaceSymbol.ConstituentNamespaces
+        {
+            get
+            {
+                return _underlying.ConstituentNamespaces.GetPublicSymbols();
+            }
+        }
+
+        IEnumerable<INamespaceOrTypeSymbol> INamespaceSymbol.GetMembers()
+        {
+            foreach (var n in _underlying.GetMembers())
+            {
+                yield return ((Symbols.NamespaceOrTypeSymbol)n).GetPublicSymbol();
```

Comment:
Are the yields here to try and avoid realizing the public symbols unless requested?

Reply:
If so, consider leaving an explanatory comment.

---
In reply to: [338829576](https://github.com/dotnet/roslyn/pull/39498#discussion_r338829576) [](ancestors = 338829576)

Reply:
> Are the yields here to try and avoid realizing the public symbols unless requested?

No. To avoid an array allocation. Does this really need a comment?

---
In reply to: [338829681](https://github.com/dotnet/roslyn/pull/39498#discussion_r338829681) [](ancestors = 338829681,338829576)

Reply:
No, that's fine. I just wanted to make sure I wasn't overthinking things.

---
In reply to: [338830593](https://github.com/dotnet/roslyn/pull/39498#discussion_r338830593) [](ancestors = 338830593,338829681,338829576)